### PR TITLE
Make use of new Egui Reactive mode

### PR DESCRIPTION
**Pull request by dragonnn**
This PR adds support for https://github.com/Smithay/smithay-egui/commit/24f7c2b06490a8fff27dd873571c37ea59cc9f3e
Egui will now report damage only if a input event is processed by it or a widget calls `ui.ctx().request_repaint();`
Currently Logger, FPS and Text widget are making use of `request_repaint`, this is also included in this PR

### CODE REVIEW

File: src/config/outputs/shell/logger/drain.rs [hunk 1]
```diff
@@ -15,8 +15,8 @@ use lazy_static::lazy_static;
 const BUFFER_SIZE: usize = 32;
 
 lazy_static! {
-    pub static ref BUFFER: Mutex<VecDeque<(Level, String)>> =
-        Mutex::new(VecDeque::with_capacity(BUFFER_SIZE));
+    pub static ref BUFFER: Mutex<(bool, VecDeque<(Level, String)>)> =
```
Comment (PolyMeilex) [inline src/config/outputs/shell/logger/drain.rs]: Could we add a comment here about what this bool is, or ideally move it to struct and let the field name describe it.

Comment (dragonnn) [reply src/config/outputs/shell/logger/drain.rs]: Sure, a struct is definitely better. Should be now fine

File: src/config/outputs/shell/logger/drain.rs [hunk 2]
```diff
@@ -14,9 +14,22 @@ use lazy_static::lazy_static;
...
+pub struct ShellBuffer {
+    pub buffer: VecDeque<(Level, String)>,
+    pub updated: bool,
+}
+
+impl ShellBuffer {
+    pub fn new() -> Self {
+        Self {
+            buffer: VecDeque::with_capacity(BUFFER_SIZE),
+            updated: true,
```
Comment (PolyMeilex) [inline src/config/outputs/shell/logger/drain.rs:26]: Looks great! But before you moved this to a struct, it used to be `false`, is this intentional change?

Comment (dragonnn) [reply src/config/outputs/shell/logger/drain.rs:26]: Yes, just to make sure when everything starts and for example a the only used widget is the Logger widget and logs are empty the widget gets it first draw.
Text widget also starts with true.

**Comment by PolyMeilex** [comment]
Thanks!

**Pull request merged**